src/function.py

```diff
@@ -34,18 +36,18 @@ def zip_and_upload_folder(client: CogniteClient, folder: Path, file_name: str) -
     raise FunctionDeployError("Failed to upload file to Cognite Data Fusion")
 
 
-def await_function_deployment(client: CogniteClient, external_id: str, wait_time_seconds: int):
+def await_function_deployment(client: CogniteClient, external_id: str, wait_time_seconds: int) -> Optional[Function]:
     t_end = time.time() + wait_time_seconds
-    while time.time() < t_end:
+    while time.time() <= t_end:
         function: Optional[Function] = client.functions.retrieve(external_id=external_id)
         if function is not None:
             if function.status == "Ready":
-                return True
+                return function
             if function.status == "Failed":
                 raise FunctionDeployError(function.error["trace"])
-        time.sleep(3.0)
+        time.sleep(randint(3, 15))  # nosec
 
-    return False
+    return None
```

Comment:
You can delete `return None`

Reply:
Doesn't None happen on the timeout?

Reply:
It is superfluous, the functions will return `None` on timeout

Reply:
It is superfluous, the function will return `None` on timeout

Reply:
I disagree. mypy (and most of the static analyzers) will complain that return statements is missing. Is it forgotten on purpose? or it's intended... I think that should be explicit, meaning yes, I want to return None if I did not get anything after awaiting.

Reply:
...but... it... is so ugly to look at 🤮  (also, mypy... don't get me started..)